targets/unity-v2/source/ExampleTestProject/Assets/Testing/Tests/Shared/Uunit/UUnitIncrementalTestRunner.cs

```diff
@@ -75,6 +75,12 @@ private void OnSuiteFinish()
 #else
         private void OnSuiteFinish()
         {
+            var report = suite.GetInternalReport();
+            if(report.failures > 0)
+            {
+                throw new Exception("Tests have failed! Ending our tests early, see this Test Summary\n" + suite.GenerateTestSummary());
```

Comment:
PostTestResultsToCloudScript is how the tests leave the device and get back to the Test Coordinator (ADO or Jenkins).
If you throw before it can do that, then you can't ever get the results back.

Reply:
It shortcuts certain issues that we have had with cloudscript (upload limits, hanging waiting for builds that never made it up there etc.) 

But to ensure we maintain current behavior, we can move this after both the login success/failure if they happen.

Reply:
I accept that the Cloud Script based solution isn't perfect, but we shouldn't make the problem worse by introducing more ways for it to fail. Intermediate solutions shouldn't make the end-goal harder to achieve.